aptos-move/block-executor/src/executor.rs

```diff
@@ -478,9 +481,10 @@ where
             return Ok(vec![]);
         }
 
-        let num_txns = signature_verified_block.len() as u32;
-        let last_input_output = TxnLastInputOutput::new(num_txns);
-        let scheduler = Scheduler::new(num_txns);
+        let num_txns = signature_verified_block.len();
+        let positions_by_txn_idx = IndexMapping::inverse(block_size, &txn_indices);
```

Comment:
because of the clone cost.
There are 2 big collections now: `txn_indices` and `positions_by_txn_idx`.
`TxnLastInputOutput` only needs `positions_by_txn_idx`.
`Scheduler` needs both.

Reply:
That's okay, these are primitive types, so cloning them shouldn't be too much overhead. I would prefer the code being more readable at the cost of this slight performance cost :).

Reply:
fixed.